django/forms/boundfield.py

```diff
@@ -267,7 +267,7 @@ def template_name(self):
 
     @property
     def id_for_label(self):
-        return 'id_%s_%s' % (self.data['name'], self.data['index'])
+        return self.data['attrs'].get('id')
```

Comment:
Hey @prestontimmons — I don't suppose you have half-a-cycle to glance at this do you? 

This is unchanged since you added `BoundWidget` in https://github.com/django/django/commit/b52c73008a9d67e9ddbb841872dc15cdd3d6ee01#diff-9da17cc8ef134b4f62a6cd6df0ff75a0d970737818bf54ab51adddfb1460ceb7R292

In some cases (`test_iterable_boundfield_select`, `test_multiple_choice_checkbox`, `test_form_with_noniterable_boundfield,) the `data` passed to `BoundWidget` doesn't include the `id` — but that seems OK... — in those cases we're not outputting the `id` for the individual subwidget. …

I'm just wondering if you had a specific cases in mind implementing this originally…?  🤔

If no bells ring, that's cool too. :) 
Thanks.

Reply:
Hi Carlton. I don't recall, but looking now it seems like `id_for_label` wasn't necessary. The `input_option` template uses `widget.attrs.id`, so it makes sense for this too, also.

https://github.com/django/django/blob/main/django/forms/templates/django/forms/widgets/input_option.html

If the existing tests are passing then this change looks okay to me. The rendering here should match the output given when rendering the widget directly:

https://github.com/django/django/blob/6fb6bb5b081350f7d68e85ddac4b9f46364aa6e3/tests/forms_tests/tests/test_forms.py#L892

Reply:
Thanks @prestontimmons — that's really handy. I appreciate it was a long time ago now, but just wanted to see that no immediate red flag popped up for you. (The discussion on #6498 is a monster 😅)